src/_generic/features.ts

```diff
@@ -0,0 +1,15 @@
+import { ApiInstance, FilterParams } from './types';
+import { createQueryObject } from '../util';
+import { GotPromise } from 'got';
+
+// eslint-disable-next-line @typescript-eslint/no-explicit-any
+export const listFeature = (apiUrl: string) => (api: ApiInstance) => (query?: FilterParams): GotPromise<any> =>
```

Comment:
Wouldn't it be better if we flipped the first two arguments? first we would get the api instance and then the apiUrl. This would follow the principle of putting the arguments that are more dynamic last, allowing us to reuse and compose more easily.
The API object will certainly not change and can be reused, but the apiUrl will change for each different feature that we list/get/create.
What do you think?

Reply:
Of course, it would impact the index.ts file (e.g. customer/index.ts) because we no longer would be able to pass the API URL directly to it.

Reply:
I explain this to victor. Here we are using feature only to HTTP, but in a case that our ApiInstance turn on a PersistInterface (that talks with API or Database), apiUrl it'll be something like "namespace".

So first argument will be used inside namespace to give to API or any Instance a reference of request direction.